Kernels/tides_kernels.py

```diff
@@ -0,0 +1,721 @@
+    A_Vn2 = f_n2*fieldset.Van2[t, particle.depth, lat3, lon3]
+    A_Vk1 = f_k1*fieldset.Vak1[t, particle.depth, lat3, lon3]
+    A_Vo1 = f_o1*fieldset.Vao1[t, particle.depth, lat3, lon3]
+    A_Vq1 = f_q1*fieldset.Vaq1[t, particle.depth, lat3, lon3]
+    A_Vm4 = f_m4*fieldset.Vam4[t, particle.depth, lat3, lon3]
+    A_Vk2 = f_k2*fieldset.Vak2[t, particle.depth, lat3, lon3]
+    A_Vp1 = f_p1*fieldset.Vap1[t, particle.depth, lat3, lon3]
+    A_Vmf = f_mf*fieldset.Vamf[t, particle.depth, lat3, lon3]
+    A_Vmm = f_mm*fieldset.Vamm[t, particle.depth, lat3, lon3]
+
+    #phase meridional 
+    ph_Vm2 = fieldset.omegam2*(t + fieldset.t0rel) + fieldset.V_m2 + u_m2 - fieldset.Vgm2[t, particle.depth, lat3, lon3]
+    ph_Vs2 = fieldset.omegas2*(t + fieldset.t0rel) + fieldset.V_s2 + u_s2 - fieldset.Vgs2[t, particle.depth, lat3, lon3]
+    ph_Vn2 = fieldset.omegan2*(t + fieldset.t0rel) + fieldset.V_n2 + u_n2 - fieldset.Vgn2[t, particle.depth, lat3, lon3]
+    ph_Vk1 = fieldset.omegak1*(t + fieldset.t0rel) + fieldset.V_k1 + u_k1 - fieldset.Vgk1[t, particle.depth, lat3, lon3]
+    ph_Vo1 = fieldset.omegao1*(t + fieldset.t0rel) + fieldset.V_o1 + u_o1 - fieldset.Vgo1[t, particle.depth, lat3, lon3]
+    ph_Vq1 = fieldset.omegaq1*(t + fieldset.t0rel) + fieldset.V_q1 + u_q1 - fieldset.Vgq1[t, particle.depth, lat3, lon3]
+    ph_Vm4 = fieldset.omegam4*(t + fieldset.t0rel) + fieldset.V_m4 + u_m4 - fieldset.Vgm4[t, particle.depth, lat3, lon3]
+    ph_Vk2 = fieldset.omegak2*(t + fieldset.t0rel) + fieldset.V_k2 + u_k2 - fieldset.Vgk2[t, particle.depth, lat3, lon3]
+    ph_Vp1 = fieldset.omegap1*(t + fieldset.t0rel) + fieldset.V_p1 + u_p1 - fieldset.Vgp1[t, particle.depth, lat3, lon3]
+    ph_Vmf = fieldset.omegamf*(t + fieldset.t0rel) + fieldset.V_mf + u_mf - fieldset.Vgmf[t, particle.depth, lat3, lon3]
+    ph_Vmm = fieldset.omegamm*(t + fieldset.t0rel) + fieldset.V_mm + u_mm - fieldset.Vgmm[t, particle.depth, lat3, lon3]
+
+    #calculate velocity due to all tidal constituents
+    U4tide = (
+        + A_Um2*math.cos(ph_Um2) 
+        + A_Us2*math.cos(ph_Us2) 
+        + A_Un2*math.cos(ph_Un2) 
+        + A_Uk1*math.cos(ph_Uk1) 
+        + A_Uo1*math.cos(ph_Uo1) 
+        + A_Uq1*math.cos(ph_Uq1) 
+        + A_Um4*math.cos(ph_Um4) 
+        + A_Uk2*math.cos(ph_Uk2)
+        + A_Up1*math.cos(ph_Up1)
+        + A_Umf*math.cos(ph_Umf)
+        + A_Umm*math.cos(ph_Umm) 
+    )
+
+    V4tide = (
+        + A_Vm2*math.cos(ph_Vm2)
+        + A_Vs2*math.cos(ph_Vs2)
+        + A_Vn2*math.cos(ph_Vn2)
+        + A_Vk1*math.cos(ph_Vk1)
+        + A_Vo1*math.cos(ph_Vo1)
+        + A_Vq1*math.cos(ph_Vq1)
+        + A_Vm4*math.cos(ph_Vm4)
+        + A_Vk2*math.cos(ph_Vk2)
+        + A_Vp1*math.cos(ph_Vp1)
+        + A_Vmf*math.cos(ph_Vmf)
+        + A_Vmm*math.cos(ph_Vmm)
+      )
+    
+    particle_dlon += (U1tide + 2*U2tide + 2*U3tide + U4tide)/6. * particle.dt
+    particle_dlat += (V1tide + 2*V2tide + 2*V3tide + V4tide)/6. * particle.dt
```

Comment:
```suggestion
    particle_dlat += (V1tide + 2*V2tide + 2*V3tide + V4tide)/6. * particle.dt

```
Good practice to add a return at the end of a file